ci/install_dependencies.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+set -e
+
+# this script expects one parameter, which is the path to utils.source
+# if the parameter is omitted, it defaults to the relative path in the
+# repository, but it can be provided manually (e.g. for vagrant VMs)
+
+if [ $# -eq 0 ]; then
+    source $(dirname "${BASH_SOURCE[0]}")/../contrib/vms/utils.source
+elif [ $# -gt 1 ]; then
+    cat << EOF
+usage: install_dependencies.sh [path/to/utils.source]
+
+Installs the dependencies required to build & test exiv2 on some Linux
+distributions.
+The optional parameter is the path to the utils.source file, which can be set to
+an alternative location (currently used for vagrant builds)
+EOF
+    exit 1
+else
+    source "$1"
+fi
+
+distro_id=$(get_distro_id)
+
+case "$distro_id" in
+    'fedora')
+        dnf -y --refresh install gcc-c++ clang cmake make ccache expat-devel zlib-devel libssh-devel libcurl-devel gtest-devel which dos2unix
```

Comment:
[question] I noticed that you are installing gcc and clang in all the different distros. Are you using both in gitlab?

Reply:
Yes, I am. Gitlab builds every combination of gcc/clang, Release/Debug and shared/static library.